From 7bc7db0c8eba5a44142287c855a4be98c9f9cf1a Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:12:27 -0400
Subject: [ethernet] igb: use mac loopback for i354 backplane

Message-id: <1409062359-13063-70-git-send-email-sassmann@redhat.com>
Patchwork-id: 89186
O-Subject: [RHEL7.1 PATCH 69/81] igb: use mac loopback for i354 backplane
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Todd Fujinaka <todd.fujinaka@intel.com>

We can't know what PHY is to be used for i354 backplane, so use MAC
loopback for ethtool tests.

Signed-off-by: Todd Fujinaka <todd.fujinaka@intel.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 3cfcf036e1d0641ace932c2bd01211d8254a9291)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_ethtool.c b/drivers/net/ethernet/intel/igb/igb_ethtool.c
index 3889c20..e052cf2 100644
--- a/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ b/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -1665,8 +1665,8 @@ static int igb_setup_loopback_test(struct igb_adapter *adapter)
   (hw->device_id == E1000_DEV_ID_DH89XXCC_SERDES) ||
   (hw->device_id == E1000_DEV_ID_DH89XXCC_BACKPLANE) ||
   (hw->device_id == E1000_DEV_ID_DH89XXCC_SFP) ||
-  (hw->device_id == E1000_DEV_ID_I354_SGMII)) {
-
+  (hw->device_id == E1000_DEV_ID_I354_SGMII) ||
+  (hw->device_id == E1000_DEV_ID_I354_BACKPLANE_2_5GBPS)) {
    /* Enable DH89xxCC MPHY for near end loopback */
    reg = rd32(E1000_MPHY_ADDR_CTL);
    reg = (reg & E1000_MPHY_ADDR_CTL_OFFSET_MASK) |
-- 
1.7.1